providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/OperatorTest.java

```diff
@@ -33,6 +35,46 @@ void flagKeyPresent() throws TargetingRuleException {
         assertEquals(true, evalVariant);
     }
 
+    @Test
+    void timestampPresent() throws TargetingRuleException {
+        // given
+
+        // rule asserting $flagd.timestamp is a number (i.e., a Unix timestamp)
+        final String targetingRule = "{\"var\":[\"$flagd.timestamp\"]}";
+
+        // when
+        Object timestampString = OPERATOR.apply("some-key", targetingRule, new ImmutableContext());
+
+        long timestamp = (long) Double.parseDouble(timestampString.toString());
+
+        // generating current unix timestamp & 5 minute threshold
+        long currentTimestamp = Instant.now().getEpochSecond();
+        long thresholdPast = currentTimestamp - (5);
+        long thresholdFuture = currentTimestamp + (5);
+
+        // checks if the timestamp is within 5 minutes of the current time
+        assertTrue(timestamp >= thresholdPast && timestamp <= thresholdFuture);
+    }
+
+    @Test
+    void testExtractSubPropertyFromFlagd() {
+        // Given
+        Map<String, Object> flagdProperties = new HashMap<>();
+        flagdProperties.put(Operator.FLAG_KEY, "some-key");
+        flagdProperties.put(Operator.TIME_STAMP, 1634000000L); // Changed to long
+
+        Map<String, Object> dataMap = new HashMap<>();
+        dataMap.put(Operator.FLAGD_PROPS_KEY, flagdProperties);
+
+        // When
+        Object flagKey = Operator.FlagProperties.extractSubPropertyFromFlagd(dataMap, Operator.FLAG_KEY);
+        Object timestamp = Operator.FlagProperties.extractSubPropertyFromFlagd(dataMap, Operator.TIME_STAMP);
```

Comment:
Testing `extractSubPropertyFromFlagd` is less important than testing the constructor (the constructor is what's actually used in code).

All you have to do is slightly modify this test to instead construct a `FlagProperties`, instance, passing it `dataMap` and make sure it's members are what you'd expect. That will indirectly test `extractSubPropertyFromFlagd`, which can just be a private method, used in the constructor.